.github/workflows/ci.yml

```diff
@@ -46,7 +46,7 @@ env:
         secrets.CONSTRAINTS_GITHUB_REPOSITORY || 'apache/airflow' }}
   # In builds from forks, this token is read-only. For scheduler/direct push it is WRITE one
   GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-  ENABLE_TEST_COVERAGE: "${{ github.event_name == 'push' }}"
+  ENABLE_TEST_COVERAGE: "true"
```

Comment:
I think this is "problematic". The reason why we do not run coverage for PRs is that we have "selective checks" - basically every PR will run a different subset of tests. If you only change google provider - only tests for core + provider + all provider that depends on google are run, if you only change "alibaba" - only alibaba provider tests are run. If you only change "cli" - only CLi tests are run, and so one. Which gives enormous speed boost for vast majority of PRs. 

I think if we allow coverage for "pull_request" type of events it will get even more "bumpy".

Reply:
That's why currently the coverage tests are only enabled for "push" runs - i.e. those are the "merges" that manage to complete all the tests.

Reply:
It just workaround to check it into the branch is it work or not, until merge to the master. I will revert this commit https://github.com/apache/airflow/pull/33605/commits/d562e629612ddd9252799fe014c8c5b5d48b05cb when finish

Reply:
Before
```
[2023-08-21T22:48:22.608Z] ['info'] => Found 7 possible coverage files:
  coverage-API-sqlite.xml
  coverage-Always-sqlite.xml
  coverage-CLI-sqlite.xml
  coverage-Core-sqlite.xml
  coverage-Other-sqlite.xml
  coverage-Providers-sqlite.xml
  coverage-WWW-sqlite.xml
```

With 6ab76c7
```
[2023-08-22T09:56:59.500Z] ['info'] => Found 9 possible coverage files:
  coverage-API-sqlite-kvwxEQHj.xml
  coverage-Always-sqlite-rQoz8s85.xml
  coverage-CLI-sqlite-OPUreDDO.xml
  coverage-Core-sqlite-xwC5zEmr.xml
  coverage-Other-sqlite-p3tUHgcp.xml
  coverage-Providers-sqlite-GXp1Lb13.xml
  coverage-Providers-sqlite-NA4wWAjB.xml
  coverage-Providers-sqlite-TAi6XAr9.xml
  coverage-WWW-sqlite-8CchlrRV.xml
```

And coverage goes back to April 2023, and become more [realistic](https://app.codecov.io/gh/apache/airflow/pull/33605/flags)

```
HEAD: 84.29%
Patch: -
Change: 14.70%
```

Reply:
Ah yes. Because I split Providers into different groups to improve parallelism. It does not have to be random (though maybe it makes sesne). `-${TEST_TYPE/\[*\]/}` - this one removed what is added to `Providers[,,,,,]` - > this is why coverage from three different provider groups have same name.

Yeah. Random suffix is probably good for coverage report.

Reply:
Very nice find :)